### PR TITLE
Added error handling (#292)

### PR DESCRIPTION
**Pull request by mgordel**

### CODE REVIEW

File: yajsapi/mid-level-api/activity/activity.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Results, BatchResults, StreamResults } from "./results";
-import { ActivityStateStateEnum } from "ya-ts-client/dist/ya-activity/src/models/activity-state";
+import { ActivityStateStateEnum as ActivityStateEnum } from "ya-ts-client/dist/ya-activity/src/models/activity-state";
```
Comment (shadeofblue) [inline yajsapi/mid-level-api/activity/activity.ts:2]: where does that original name (`ActivityStateStateEnum`) come from???

File: yajsapi/mid-level-api/activity/activity.ts [hunk 2]
```diff
@@ -104,26 +109,88 @@ export class Activity {
...
       .catch((error) => this.logger?.warn(`Got API Exception when destroying activity ${this.id}: ${error}`));
-    if (this.stateFetchIntervalId) clearInterval(this.stateFetchIntervalId);
-    await this.getState();
     if (error) this.logger?.debug("Activity ended with an error: " + error);
     else this.logger?.debug("Activity ended");
   }
 
   private async handleError(error, cmdIndex, retryCount, maxRetries) {
-    // todo
+    if (this.isTimeoutError(error)) {
```
Comment (shadeofblue) [inline yajsapi/mid-level-api/activity/activity.ts:134]: plus, shouldn't this `handleError` logic be pulled somewhere outside of the activity module? it seems generic enough, doesn't it?

File: yajsapi/mid-level-api/activity/activity.ts [hunk 3]
```diff
@@ -104,26 +109,88 @@ export class Activity {
...
+      this.logger?.warn("API request timeout." + error.toString());
+      return retryCount;
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply yajsapi/mid-level-api/activity/activity.ts:136]: we're okay because we still have (or should have) a timeout in the outer scope

**Review by shadeofblue — APPROVED** [review]
looks good but please see my comments

File: yajsapi/mid-level-api/activity/activity.ts [hunk 4]
```diff
@@ -104,26 +109,88 @@ export class Activity {
...
 
-  async getState(): Promise<ActivityStateStateEnum> {
-    const { data } = await this.stateApi.getActivityState(this.id);
-    // TODO: catch and check error
-    if (data?.state?.[0] && data?.state?.[0] !== this.state) {
-      this.state = data.state[0];
+  async getState(): Promise<ActivityStateEnum> {
+    try {
+      const { data } = await this.stateApi.getActivityState(this.id);
+      if (data?.state?.[0] && data?.state?.[0] !== this.state) {
```
[the remark this answers is not in the document]
Comment (mgordel) [reply yajsapi/mid-level-api/activity/activity.ts:115]: this was after the activity module emitted a state change. Now it's unnecessary, I'll delete it

File: yajsapi/mid-level-api/activity/activity.ts (hunk 2 above)
Comment (mgordel) [reply yajsapi/mid-level-api/activity/activity.ts:134]: It is very possible .. but I will wait with it until I find the same case in the code.

**Pull request merged**